editor/src/clj/editor/properties_view.clj

```diff
@@ -82,16 +82,25 @@
                                                      (ui/run-later (.selectAll t))))
                                                  (ui/user-data! t ::selection-at-focus nil)))))
 
-(defmulti customize! (fn [control update-fn] (class control)))
+(defmulti customize! (fn [control _ _] (class control)))
 
-(defmethod customize! TextField [^TextField t update-fn]
+(defmethod customize! TextField [^TextField t update-fn cancel-fn]
   (doto t
     (GridPane/setHgrow Priority/ALWAYS)
     (ui/on-action! update-fn)
     (ui/auto-commit! update-fn)
-    (select-all-on-click!)))
-
-(defmethod customize! TextArea [^TextArea t update-fn]
+    (select-all-on-click!)
+    (ui/bind-key! "Esc" (fn []
+                          (cancel-fn t)
+                          (when-let [parent (.getParent t)]
+                            (.requestFocus parent))))
```

Comment:
Cancel input and unselect the field: 

https://github.com/defold/defold/assets/2209596/59744a3f-87d9-418e-9d3a-4098ef9d7735